Mathlib/AlgebraicGeometry/Morphisms/ClosedImmersion.lean

```diff
@@ -133,6 +131,15 @@ theorem of_comp {X Y Z : Scheme} (f : X ⟶ Y) (g : Y ⟶ Z) [IsClosedImmersion
     simp_rw [Scheme.stalkMap_comp] at h
     exact Function.Surjective.of_comp h
 
+instance {X : Scheme.{u}} (x) : IsClosedImmersion (Spec.map (X.residue x)) :=
+  IsClosedImmersion.spec_of_surjective (X.residue x)
+    Ideal.Quotient.mk_surjective
+
+instance {X : Scheme.{u}} (x : X) : IsPreimmersion (X.fromSpecResidueField x) := by
+  dsimp only [Scheme.fromSpecResidueField]
+  rw [IsPreimmersion.comp_iff]
+  infer_instance
+
```

Comment:
I think this can go into `ResidueField` along with `SpecToEquivOfField`.

Reply:
Not sure, this depends on the `IsClosedImmersion` instance of `Spec.map (X.residue x)` and we can't import `IsClosedImmersion` in `ResidueField`. The alternative would be to add `IsPreimmersion (Spec.map (X.residue x))` in `ResidueField` separately and only later upgrade it to `IsClosedImmersion`, but not sure if this is really better than the current version.

Reply:
I feel that the latter makes more sense to me. As this result itself has little to do with closed immersions.

Reply:
Okay, I did that and added a comment on the `IsPreimmersion` instance to find the stronger result.